lib/checkleakautovar.cpp

```diff
@@ -342,6 +342,18 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
                     varAlloc.type = f->groupId;
                     varAlloc.status = VarInfo::ALLOC;
                 }
+                const Library::AllocFunc* g = mSettings->library.realloc(tokRightAstOperand->previous());
+                if (g && g->arg == -1) {
+                    VarInfo::AllocInfo& varAlloc = alloctype[varTok->varId()];
+                    varAlloc.type = g->groupId;
+                    varAlloc.status = VarInfo::ALLOC;
```

Comment:
shouldn't the input be allocated already?

Reply:
It's the output that is set (it's been refactored and is hopefully less confusing now).